### PR TITLE
Set-up grading table to allow for batch grading

### PR DESCRIPTION
**Pull request by rrtheonlyone**
The grading table needs some tweaks to allow for batch grading. 

The PR adds:
- pagination (50 entries per page)
- checkbox to show all submissions

### CODE REVIEW

File: src/sagas/backend.ts [hunk 1]
```diff
@@ -406,8 +417,11 @@ async function postAssessment(id: number, tokens: Tokens): Promise<Response | nu
  * GET /grading
  * @returns {Array} GradingOverview[]
  */
-async function getGradingOverviews(tokens: Tokens): Promise<GradingOverview[] | null> {
-  const response = await request('grading', 'GET', {
+async function getGradingOverviews(
+  tokens: Tokens,
+  group = false
```
Comment (remo5000) [inline src/sagas/backend.ts]: Could this be a parameter with no default value? It's safer behaviour, and helps with figuring out issues.

Comment (remo5000) [reply src/sagas/backend.ts]: Just noticed that you use this function at both `fetchGrading` and `fetchGradingOverview`, that's nice

File: src/actions/session.ts [hunk 2]
```diff
@@ -34,6 +34,10 @@ export const fetchGradingOverviews = () => ({
   type: actionTypes.FETCH_GRADING_OVERVIEWS
 })
 
+export const fetchGradingOverviewsForGroup = () => ({
```
[the remark this answers is not in the document]
Comment (rrtheonlyone) [reply src/actions/session.ts]: Yup your right! 

I took away this completely and put it as a parameter

File: src/components/academy/grading/index.tsx [hunk 3]
```diff
@@ -24,6 +24,7 @@ import { OwnProps as GradingWorkspaceProps } from './GradingWorkspace'
 type State = {
   columnDefs: ColDef[]
   filterValue: string
+  groupsFilter: boolean
```
[the remark this answers is not in the document]
Comment (rrtheonlyone) [reply src/components/academy/grading/index.tsx]: Done!

File: src/components/academy/grading/index.tsx [hunk 4]
```diff
@@ -187,6 +206,17 @@ class Grading extends React.Component<IGradingProps, State> {
     }
   }
 
+  private handleGroupsFilter = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const checkStatus = event.target.checked
+    this.setState({ groupsFilter: checkStatus })
+
+    if (checkStatus) {
```
[the remark this answers is not in the document]
Comment (rrtheonlyone) [reply src/components/academy/grading/index.tsx]: Yup exactly :)

File: src/sagas/backend.ts (hunk 1 above)
Comment (rrtheonlyone) [reply src/sagas/backend.ts]: Haha refactored as well 

(Although I was forced to use default values for my action to use `ContentDisplay`)

**Pull request merged**